dialogues/risawoz/src/convert.py

```diff
@@ -0,0 +1,300 @@
+            dialogue_id = dialogue["dialogue_id"]
+            scenario = {
+                "UserTask": dialogue.get("goal", ""),
+                "WizardCapabilities": [{"Task": domain} for domain in dialogue["domains"]],
+            }
+            events = []
+            for turn in dialogue["dialogue"]:
+                user_turn_event = build_user_event(turn)
+                if "db_results" in turn and turn["db_results"]:
+                    wizard_query_event = build_wizard_event(turn, mode="query")
+                    kb_event = build_kb_event(wizard_query_event, mongodb_host, api_map)
+                    wizard_normal_event = build_wizard_event(turn, mode="normal")
+                    events += [user_turn_event, wizard_query_event, kb_event, wizard_normal_event]
+                else:
+                    wizard_event = build_wizard_event(turn)
+                    events += [user_turn_event, wizard_event]
+            processed_data[key][dialogue_id] = {"Dialogue_id": dialogue_id, "Scenario": scenario, "Events": events}
+    processed_data = dict(processed_data)
+    return processed_data
+
+
+def build_mock_pred_data(test_data_path):
+    with open(test_data_path, 'r') as f:
+        test_data = json.load(f)
+    fake_pred_data = {}
+    for dialog in test_data:
+        fake_pred_dialog = {"turns": {}}
+        final_dialog_state = {}
+        for turn in dialog["dialogue"]:
+            fake_pred_dialog["turns"][str(turn["turn_id"] + 1)] = {"response": [turn["system_utterance"]]}
+            # build dialog state
+            dialog_state = {}
+            for ds, v in turn["belief_state"]["inform slot-values"].items():
+                v = ''.join(v.split())
+                d, s = ds.split("-")
+                if d in dialog_state.keys():
+                    if s in dialog_state[d].keys():
+                        dialog_state[d][s]["value"].append(v)
+                    else:
+                        dialog_state[d][s] = {}
+                        dialog_state[d][s]["relation"] = "equal_to"
+                        dialog_state[d][s]["value"] = [v]
+                else:
+                    dialog_state[d] = {}
+                    dialog_state[d][s] = {}
+                    dialog_state[d][s]["relation"] = "equal_to"
+                    dialog_state[d][s]["value"] = [v]
+            fake_pred_dialog["turns"][str(turn["turn_id"] + 1)]["state"] = dialog_state
+            # update final dialog state in each turn and build turn API text
+            turn_api_text = []
+            for d, sv in dialog_state.items():
+                if d in final_dialog_state.keys():
+                    final_dialog_state[d].update(sv)
+                else:
+                    final_dialog_state[d] = sv
+                # build turn API text
+                try:
+                    if len(turn["db_results"]) > 1:
+                        first_result = eval(turn["db_results"][1].replace("true", "True").replace("false", "False"))
```

Comment:
In bitod, the agent only needs the first result to continue the conversation. In other words, they rank the results based on user rating and return the first. This way you always get the same result (output) for the same constraints (input). Is that also the case in Risawoz? Does agent always look at the first result to formulate its response?

Reply:
Yes, the first result is always used in the conversation.